cos_alerter/alerter.py

```diff
@@ -151,6 +160,42 @@ def initialize():
                 "notify_time": None,
             }
 
+        # Recover any state that was dumped on last exit.
+        if config["state_file"].exists():
+            with config["state_file"].open() as f:
+                existing_clients = json.load(f)
+            config["state_file"].unlink()
+            for client in existing_clients:
+                if client in state["clients"]:
+                    state["clients"][client]["alert_time"] = existing_clients[client]["alert_time"]
+                    state["clients"][client]["notify_time"] = existing_clients[client][
+                        "notify_time"
+                    ]
+
+    # This is difficult to test in unit tests because it acquires and does not release all of the
+    # locks. When integration tests have been solved we need to remove the "no cover" from this
+    # method.
+    @staticmethod
+    def dump_and_pause():  # pragma: no cover
+        """Dump the state of the program and exit gracefully.
+
+        This function acquires all the locks and never releases them, affectively pausing the
```

Comment:
```suggestion
        This function acquires all the locks and never releases them, effectively pausing the
```

although the affective pause has a cool Freudian ring to it